### ISSUE TITLE
hasLayerNorm throws Exceptions on SimpleRnn

### ISSUE THREAD

**Issue by thomas-trendsoft** [deeplearning4j/deeplearning4j#9888]
I tried to activate layer normalization on a SimpleRnn. 

`
					.addLayer("enc_ffn" + i, new SimpleRnn.Builder()
							.dropOut(rate)
							.hasLayerNorm(true)
							.activation(Activation.RELU)
							.nIn(EMB_SIZE)
							.nOut(EMB_SIZE)
							.build(), "out" + i)
`

Please describe our issue, along with:
- expected the layer to feed forward like without .hasLayerNorm (works wihtout layerNorm as expected)
- encountered behavior

Error at [/Users/runner/work/deeplearning4j/deeplearning4j/libnd4j/include/ops/declarable/generic/nn/layer_norm.cpp:47:0]:
LAYER_NORM OP: wrong shape of gain array, expected is {128}, but got [1, 128] instead !
[main] ERROR org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner - Failed to execute op layer_norm. Attempted to execute with 3 inputs, 1 outputs, 0 targs,1 bargs and 1 iargs. Inputs: [(FLOAT,[128,128],f), (FLOAT,[1,128],f), (FLOAT,[1,128],f)]. Outputs: [(FLOAT,[128,128],f)]. tArgs: -. iArgs: [1]. bArgs: [true]. Op own name: "36003208-19ee-442a-8d0a-b868e81d163b" - Please see above message (printed out from c++) for a possible cause of error.
[main] WARN org.deeplearning4j.earlystopping.trainer.BaseEarlyStoppingTrainer - Early stopping training terminated due to exception at epoch 0, iteration 0
java.lang.RuntimeException: Op [layer_norm] execution failed
	at org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner.exec(NativeOpExecutioner.java:1561)
	at org.deeplearning4j.nn.layers.recurrent.SimpleRnn.activateHelper(SimpleRnn.java:263)
	at org.deeplearning4j.nn.layers.recurrent.SimpleRnn.activate(SimpleRnn.java:213)
	at org.deeplearning4j.nn.graph.vertex.impl.LayerVertex.doForward(LayerVertex.java:110)
	at org.deeplearning4j.nn.graph.ComputationGraph.ffToLayerActivationsInWS(ComputationGraph.java:2139)
	at org.deeplearning4j.nn.graph.ComputationGraph.computeGradientAndScore(ComputationGraph.java:1376)
	at org.deep

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j Versions: M2.1
* Platform information: Mac and Linux
* CUDA version none CPU OpenBLAS

**Comment by agibsonccc** [comment]
@thomas-trendsoft do you mind posting a full reproducing network? I'd need to see the context in which this runs to even determine if this is a bug on our side or not. It likely is looking from the stack trace. If I had to guess it's probably a simple left over issue from migrating from only 2d arrays to allowing more flexibility in our vectors.

We have multiple other more up to date implementations of rnns (lstms, grus,  bidrectionalrnns, static rnns,..)
exposed through samediff. This older api is largely going to be kept as is minus bug fixes for keras import and for folks who just want to train a network through a simpler builder pattern.

 While I will fix this, I just want you to be aware of that and why simpler things like this were missed. Apologies.

**Comment by agibsonccc** [comment]
Closing — this issue is over 2 years old with no recent activity or reproduction steps. The underlying code has been significantly rewritten. If you still experience this issue with a recent version, please open a new issue.
